### PR TITLE
🐛 workspace.zip disappears while uploading

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

The dynamic sidecar concurrently uploads the output port files and the workspace file.
An issue was identified inside the dynamic sidecar:
- the output port archiver was deleting the entire ```/tmp``` folder after the upload of the output ports
- since the upload of workspace is happening concurrently, and the ```workspace.zip``` file was located in this very ```/tmp``` folder it would be deleted, and the logs were correctly reporting that the file was not there!

This PR fixes the archiver of output ports by using the temporary directory context manager.

--> Always use context managers

## Related issue/s

fixes https://github.com/ITISFoundation/osparc-issues/issues/664

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
This was a great pair-review with very useful outcome! 🎉 
thx!

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py [hunk 1]
```diff
@@ -78,77 +79,68 @@ async def upload_outputs(
...
+                    files_and_folders_list[0].is_file()
+                    or files_and_folders_list[0].is_symlink()
+                ):
+                    # special case, direct upload
+                    ports_values[port.key] = files_and_folders_list[0]
+                    continue
+
+                # generic case let's create an archive
+                # only the filtered out files will be zipped
+                tmp_folder = await stack.enter_async_context(AioTemporaryDirectory())
```
Comment (GitHK) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py:112]: I see the culprit here. nice!

**Review by GitHK — APPROVED** [review]
👍

File: packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py [hunk 2]
```diff
@@ -70,8 +71,9 @@ async def push(
             io_log_redirect_cb=io_log_redirect_cb,
         )
     # we have a folder, so we create a compressed file
-    with TemporaryDirectory() as tmp_dir_name:
-        log.info("compressing %s into %s...", file_or_folder.name, tmp_dir_name)
+    with log_catch(log), log_context(
+        log, logging.INFO, "pushing %s", file_or_folder
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py:75]: only note here, the message changed

**Review by mrnicegyu11 — APPROVED** [review]
super, thx

File: packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py (hunk 2 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py:75]: which is correct, since it is what it is doing

**Pull request merged**